internal/compile/compile.go

```diff
@@ -1005,25 +1017,41 @@ func (fcomp *fcomp) stmt(stmt syntax.Stmt) {
 					fcomp.set(lhs)
 				}
 
+			case *syntax.UnaryExpr:
+				// *x = ...
+				fcomp.expr(lhs.X)
+				fcomp.emit(DUP)
+				fcomp.emit(VALUE)
+				set = func() {
+					fcomp.setPos(lhs.OpPos) // op == STAR
+					fcomp.emit(SETVALUE)
+				}
+
 			case *syntax.IndexExpr:
 				// x[y] = ...
-				fcomp.expr(lhs.X)
+				fcomp.addr(lhs.X)
 				fcomp.expr(lhs.Y)
 				fcomp.emit(DUP2)
 				fcomp.setPos(lhs.Lbrack)
 				fcomp.emit(INDEX)
+				if resolve.AllowAddressing {
```

Comment:
It seems error-prone to emit different bytecode depending on dialect flags. An application might compile and cache code for a library that doesn't depend on this flag. The library could be loaded by different future invocations of that library in different modes.

One solution would be to include the resolve flag in the serialized header. But I think I'd prefer the simplicity of emitting these opcodes unconditionally. They're always no-ops if `AllowAddressing` is false, right? How severe is the overhead?